add pipeline examples (#1748)
nushell/nushell.github.io: lang-guide/chapters/pipelines.md

```

Also note that `complete` is special, it doesn't work with `e>|`, `o+e>|`.

## Stdio and redirection behavior examples

> Pipeline and redirection behavior can be hard to follow when they are used with subexpression, or custom commands.  Here are some examples to show stdio behavior

Reviewer: ```suggestion
Pipeline and redirection behavior can be hard to follow when they are used with subexpressions, or custom commands.  Here are some examples that show intended stdio behavior.
```

### Examples for subexpression

- (^cmd1 | ^cmd2; ^cmd3 | ^cmd4)

| Command | Stdout     | Stderr     |
| ------- | --------   | ---------- |
| cmd1    | Piped      | Terminal   |
| cmd2    | *Terminal* | Terminal   |
| cmd3    | Piped      | Terminal   |
| cmd4    | Terminal   | Terminal   |

> - (^cmd1 | ^cmd2; ^cmd3 | ^cmd4) | ^cmd5

Reviewer: I wonder if it might be good to show where the output of cmd5 goes as well but I'm not sure you can tell that without understanding what cmd5 does. e.g. does it write to file, stdout, or stderr? 

Maybe if each table had a short description saying what it demonstrates then the user would know that cmd5 writes to stdout (or whatever the example shows) and it wouldn't be needed in the table?

Author: > Maybe if each table had a short description saying what it demonstrates

I'd like to suggest leaving it there, the example command themselves only care about the behavior of cmd(1/2/3/4).  The behavior and output of cmd5 is not important.

Reviewer: I think it would be better to have a short description above each table for clarity, but we can start with what we have here if you like.

Author: Sure.  I updated a short description for these commands

| cmd3    | Piped    | Terminal |
| cmd4    | Terminal | Piped   |

- (^cmd1 | ^cmd2; ^cmd3 | ^cmd4) o+e>| ^cmd5

| Command | Stdout   | Stderr   |
| ------- | -------- | -------- |
| cmd1    | Piped    | Terminal |
| cmd2    | Terminal | Terminal |
| cmd3    | Piped    | Terminal |
| cmd4    | Piped    | Piped   |

- (^cmd1 | ^cmd2; ^cmd3 | ^cmd4) o> test.out

| Command | Stdout   | Stderr   |
| ------- | -------- | -------- |
| cmd1    | Piped    | Terminal |
| cmd2    | File | Terminal |
| cmd3    | Piped    | Terminal |
| cmd4    | File    | Terminal   |

- (^cmd1 | ^cmd2; ^cmd3 | ^cmd4) e> test.out

| Command | Stdout   | Stderr   |
| ------- | -------- | -------- |
| cmd1    | Piped    | File |
| cmd2    | Terminal | File |
| cmd3    | Piped    | File |
| cmd4    | Terminal    | File   |

- (^cmd1 | ^cmd2; ^cmd3 | ^cmd4) o+e> test.out

| Command | Stdout   | Stderr   |
| ------- | -------- | -------- |
| cmd1    | Piped    | File |
| cmd2    | File | File |
| cmd3    | Piped    | File |
| cmd4    | File    | File   |

### Examples for custom command
Given the following custom commands

```nushell
def custom-cmd [] {
    ^cmd1 | ^cmd2
    ^cmd3 | ^cmd4
}
```

> Here are some examples to show stdio behavior.  Actually the stio behavior will be the same to previous session.

Reviewer: ```suggestion
The custom command stdio behavior is the same as the previous section.
```

> You can think the body of `custom-cmd` like `(^cmd1 | ^cmd2; ^cmd3 | ^cmd4).

Reviewer: ```suggestion
In the examples below the body of `custom-cmd` is `(^cmd1 | ^cmd2; ^cmd3 | ^cmd4).
```
